rollup-operator/src/loop-manager.js

```diff
@@ -389,16 +416,98 @@ class LoopManager{
             this.timeouts.NEXT_STATE = 5000;
             // Check I am still the winner
             const deadlineReach = await this._blockDeadline(currentBlock);
-            if (deadlineReach){
+            const currentStateRoot = await this.rollupSynch.getCurrentStateRoot();
+            const currentOnchainHash = await this.rollupSynch.getMiningOnchainHash;
+
+            if (deadlineReach || BigInt(currentStateRoot) ==! this.infoCurrentBatch.batchData.getNewStateRoot() ||
```

Comment:
maybe we could check current contract variables in a separate function and return true or false
```
_checkStateRollup(){
// get contract information
// get current batch information
// compare them
// return true | false
}
```

This way we could keep simple comparison:
```
...
const checkState = await this._checkStateRollup();
if (deadlineReach || checkState){
...
}
```

Reply:
true!